src/hosted-buttons/utils.js

```diff
@@ -220,3 +225,86 @@ export const buildHostedButtonOnApprove = ({
     });
   };
 };
+
+export function getFlexDirection({
+  layout,
+}: GetFlexDirectionArgs): GetFlexDirection {
+  return { flexDirection: layout === "horizontal" ? "row" : "column" };
+}
+
+export function shouldRenderSDKButtons(
+  fundingSources: $ReadOnlyArray<FundingSources>
+): boolean {
+  return Boolean(fundingSources?.length);
+}
+
+export function getButtonColor(
+  color: Color,
+  fundingSource: FundingSources
+): Color {
+  const colorMap = {
+    gold: {
+      paypal: "gold",
+      venmo: "blue",
+      paylater: "gold",
+    },
+    blue: {
+      paypal: "blue",
+      venmo: "silver",
+      paylater: "blue",
+    },
+    black: {
+      paypal: "black",
+      venmo: "black",
+      paylater: "black",
+    },
+    white: {
+      paypal: "white",
+      venmo: "white",
+      paylater: "white",
+    },
+    silver: {
+      paypal: "silver",
+      venmo: "blue",
+      paylater: "silver",
+    },
+  };
+
+  return colorMap[color][fundingSource];
+}
+
+export function buildButtonContainer({
+  flexDirection,
+  selector,
+}: BuildButtonContainerArgs) {
+  if (typeof selector !== "string") {
```

Comment:
I started looking at how Zoid was handling accepting a sting and an html element, but I didn't find anything useful here. Since NCP team is generating the piece of code that contains the selector and it is currently a string referring to an id, I thought it would be simpler to only accept that for the hosted button experience.

Reply:
I totally forgot about `document.querySelector`. I will update function to support.

Changes can be found in this [commit](https://github.com/paypal/paypal-checkout-components/pull/2375/commits/ba8f4f64569a3b4a2276d08c108d51660e7eddab)